### PR TITLE
Adds sample code for self service registration

### PR DESCRIPTION
**Pull request by bretterer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -517,6 +517,266 @@ if err != nil {
...
+        WithClientSecret("{CLIENT_SECRET}"),        // Required for confidential clients.
+        WithIssuer("{ISSUER}"),                     // e.g. https://foo.okta.com/oauth2/default, https://foo.okta.com/oauth2/busar5vgt5TSDsfcJ0h7
+        WithScopes([]string{"openid", "profile"}),  // Must include at least `openid`. Include `profile` if you want to do token exchange
+        WithRedirectURI("{REDIRECT_URI}"),          // Must match the redirect uri in client app settings/console
+    )
+	if err != nil {
+		panic(err)
+	}
+
+	interactHandle, err := client.Interact(context.TODO())
```
Comment (laura-rodriguez) [inline README.md:536]: I guess you'll update all code samples at once when you refactor interact to return a context.

File: README.md [hunk 2]
```diff
@@ -517,6 +517,266 @@ if err != nil {
...
+		// var attribute string
+		// attribute = remediationOption.FormValues[0].Form.FormValues[0]
+
+		fmt.Printf("%+v\n", remediationOption.Form())
+		if remediationOption.Name == "enroll-profile" {
+			userProfile := []byte(`{
+				"userProfile": {
+					"lastName": "User",
+					"firstName": "Test",
+					"email": "vijet.m@gmail.com"
```
Comment (laura-rodriguez) [inline README.md:570]: Do we want to use this email?

File: README.md [hunk 3]
```diff
@@ -517,6 +517,266 @@ if err != nil {
...
+
+			response, err = remediationOption.Proceed(context.TODO(), credentials)
+
+			if err != nil {
+				panic(err)
+			}
+	    }
+	}
+
+	// fmt.Println(string(response.Raw()))
```
Comment (laura-rodriguez) [inline README.md:754]: Should we delete this line?

File: README.md [hunk 4]
```diff
@@ -517,6 +517,266 @@ if err != nil {
...
+
+	for _, remediationOption := range response.Remediation.RemediationOptions {
+
+		// var attribute string
+		// attribute = remediationOption.FormValues[0].Form.FormValues[0]
```
Comment (laura-rodriguez) [inline README.md:759]: Should we delete these lines?

Comment (laura-rodriguez) [reply README.md:759]: I notice there other commented lines above.

**Review by laura-rodriguez — APPROVED** [review]
A few minor comments, then LGTM 👍

File: README.md (hunk 1 above)
Comment (bretterer) [reply README.md:536]: Yes.  That is the plan, since this is the way it works right now

**Review by bryanapellanes-okta — APPROVED** [review]
👍

**Pull request merged**